tests/io/datasets/test_seq.py

```diff
@@ -120,12 +129,217 @@ def test_positive_sync_offset(default_seq, lt_ctx):
+                 'Rows="2048" Columns="2048"><Defect Rows="1155-1156"/><Defect ' \
+                 'Rows="1706-1707"/></BadPixelMap></BadPixels>' \
+                 '</Configuration>'
+    metadata = {
+        "UnbinnedFrameSizeX": 2048,
+        "UnbinnedFrameSizeY": 2048,
+        "OffsetX": 1024,
+        "OffsetY": 1024,
+        "HardwareBinning": 2
+    }
+    test_arr = np.zeros((1024, 1024), dtype=bool)
+    test_arr[644] = True
+    test_arr[643] = True
+    test_arr[:, 143:145] = True
+    test_arr[:, 138] = True
+    expected_res = _load_xml_from_string(xml=xml_string, metadata=metadata)
+    assert np.array_equal(expected_res.todense(), test_arr)
+
+
+def test_correct_bad_pixel_map_selector():
+    xml = '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>' \
+          '<Configuration><PixelSize></PixelSize><DiffPixelSize></DiffPixelSize>' \
+          '<BadPixels><BadPixelMap Rows="4096" Columns="4096">' \
+          '<Defect Columns="1310-1312"/></BadPixelMap><BadPixelMap ' \
+          'Rows="4126" Columns="4024"><Defect Rows="1155-1156"/><Defect ' \
+          'Rows="1706-1707"/></BadPixelMap></BadPixels>' \
+          '</Configuration>'
+    tree = ET.fromstring(xml)
+    excluded_rows_dict = xml_defect_data_extractor(tree)
+    assert excluded_rows_dict["cols"] == [['1310', '1312']]
+
+
+def test_correct_bad_pixel_map_selector_2():
+    xml = '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>' \
+          '<Configuration><PixelSize></PixelSize><DiffPixelSize></DiffPixelSize>' \
+          '<BadPixels><BadPixelMap Rows="4096" Columns="4096">' \
+          '<Defect Columns="1310-1312"/></BadPixelMap><BadPixelMap Binning="2" ' \
+          'Rows="4080" Columns="4096"><Defect Rows="1155-1156"/><Defect ' \
+          'Rows="1706-1707"/></BadPixelMap></BadPixels>' \
+          '</Configuration>'
+    tree = ET.fromstring(xml)
+    excluded_rows_dict = xml_defect_data_extractor(tree)
+    assert excluded_rows_dict["cols"] == [['1310', '1312']]
+
+
+def test_map_size():
+    xml = '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>' \
+          '<BadPixels>' \
+          '<BadPixelMap Rows="4096" Columns="4096"></BadPixelMap>' \
+          '<BadPixelMap Binning="2" Rows="4080" Columns="4096"></BadPixelMap>' \
+          '</BadPixels>'
+    tree = ET.fromstring(xml)
+    bad_pixel_maps = tree.findall('.//BadPixelMap')
+    xy_map_sizes, map_sizes = xml_map_sizes(bad_pixel_maps)
+    xy_map_sizes_expected = [(4096, 4096), (4096, 4080), (1, 2)]
+    assert xy_map_sizes == xy_map_sizes_expected
+
+
+def test_unbinned_map_maker():
+    xml = '<?xml version="1.0" encoding="UTF-8" standalone="yes"?>' \
+          '<BadPixels>' \
+          '<BadPixelMap Rows="4096" Columns="4096"></BadPixelMap>' \
+          '<BadPixelMap Rows="2048" Columns="2048"></BadPixelMap>' \
+          '<BadPixelMap Rows="2048" Columns="1024"></BadPixelMap>' \
+          '<BadPixelMap Binning="2" Rows="4080" Columns="4096"></BadPixelMap>' \
```

Comment:
Hm, why do the `BadPixelMap`s have different aspect ratios? Don't they always conform to the sensor aspect ratio?

Reply:
Same question for the other test cases below

Reply:
I was not sure if a case where hey have different aspect ratios possible so I implemented it in the xml_map_index_selector's else branch